### ISSUE TITLE
login incorectly assumes that the first argument is credentials iff there are two arguments

### ISSUE THREAD

**Issue by jteplitz602** [angular/angularfire2#71]
It should instead be using the auth method to determine this because if we're using CustomToken, Password, or OAuthToken we can just pass in the credentials object and no config. Right now `login` will reject the promise in this case but it should work.

**Comment by davideast** [comment]
I think it's fine that credentials is first, as long as we have a check for Password. Both CustomToken and OAuthToken require you to provide a `config` object as the second parameter, but Password should not.

Password currently requires you to provide a `config` object as the second parameter or it will provide a confusing error of `You must include credentials to use this auth method.`. This is because if only one param is provided, `login()` expects it to be the config, which should not be the case for Password.

Password should work with credentials as the only param:

```ts
constructor(af: AngularFire) {
  af.login({ email: 'email@email.com', password: 'secret' }).then(auth => console.log(auth));
}
```

To do this we can provide a `PasswordCredentials` interface and an extra overload:

```ts
export interface PasswordCredentials {
  email: string;
  password: string;
}
public login(credentials?: PasswordCredentials): Promise<FirebaseAuthState>;
```

Then in `.login()` when we are checking the argument length we can do as simple check on a single argument:

```ts
    if (arguments.length > 2) {
      return this._reject('Login only accepts a maximum of two arguments.');
    } else if (arguments.length == 2) {
      credentials = obj1;
      config = obj2;
    } else if (arguments.length == 1) {
      // Check if obj1 is password credentials
      if (obj1.password && obj1.email) {
        credentials = obj1;
        config = {};
      } else {
        config = obj1; 
      }
    }
```

**Comment by davideast** [comment]
Closed with #132.
